stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkEventReporterImpl.java

```diff
@@ -152,10 +152,29 @@ public void responseHeadersReceived(InspectorResponse response) {
           getResourceTypeHelper().determineResourceType(contentType) :
           Page.ResourceType.OTHER;
       receivedParams.response = responseJSON;
+      initAsyncPrettyPrinterForResponse(response, peerManager);
       peerManager.sendNotificationToPeers("Network.responseReceived", receivedParams);
     }
   }
 
+  public static void initAsyncPrettyPrinterForResponse(
+      InspectorResponse response,
+      NetworkPeerManager peerManager) {
+    AsyncPrettyPrinterRegistry registry = peerManager.getAsyncPrettyPrinterRegistry();
+    for (int i = 0, count = response.headerCount(); i < count; i++) {
+      AsyncPrettyPrinterFactory factory = registry.lookup(response.headerName(i));
+      if (factory != null) {
+        AsyncPrettyPrinter asyncPrettyPrinter = factory.getInstance(
+            response.headerName(i),
+            response.headerValue(i));
+        peerManager.getResponseBodyFileManager().associateAsyncPrettyPrinterWithId(
+            response.requestId(),
+            asyncPrettyPrinter);
+        break;
```

Comment:
This is usually considered a bad code smell and can be corrected by dividing the logic into two methods:

```java
  public static void initAsyncPrettyPrineterForResponse(...) {
    prettyPrinter = matchPrettyPrinterToResponse(...);
    if (prettyPrinter != null) {
      associateAsyncPrettyPrinter(...);
    }
  }

  @Nullable
  public static AsyncPrettyPrinter matchPrettyPrinterToResponse(...);
```

Many engineers find this easier logic to follow because they can more easily predict the exact flow control semantics of `matchPrettyPrinterToResponse(...)` based on its name and signature.